README.md

```diff
@@ -1,40 +1,85 @@
 # Fortnox.NET
+
 .NET bindings for the <a href="https://developer.fortnox.se/documentation/">Fortnox</a> API.
- 
+
 ## Usage
+
 To access the Fortnox API you need an Access Token and Client Secret: <a href="https://developer.fortnox.se/general/authentication/">https://developer.fortnox.se/general/authentication/</a>.
- 
+
 The unit tests in this repository serves as a way for us to make sure our code works as well as examples of how to call the different routes that the Fortnox API provides. There are mainly two ways to make a request using this SDK, one for requesting a list of resources and one for requesting a single resource.
- 
+
 ### Listed resource
+
 When requesting a list of resources you create a ListRequest object of the type of resource you are interested in.
- 
+
 ```CSharp
 var request = new OrderListRequest("ACCESS-TOKEN", "CLIENT-SECRET");
 var orders = await OrderService.GetOrdersAsync(request);
 ```
- 
+
 You can also add filters and ordering to routes that accepts it, for example:
- 
+
 ```CSharp
 var request = new EmployeeListRequest("ACCESS-TOKEN", "CLIENT-SECRET")
 {
     Filter = EmployeeFilters.Active
 };
 var employeeList = await EmployeeService.GetEmployeesAsync(request);
 ```
- 
+
 ### Single resource
+
 When requesting a single resource you create a FortnoxApiRequest object and pass that into the service to make your request.
- 
+
 ```CSharp
 var request = new FortnoxApiRequest("ACCESS-TOKEN", "CLIENT-SECRET");
 var response = await ArticleService.GetArticleAsync(request, "100370");
 ```
- 
+
+### WebSocket
+
+If you want to subscribe to changes instead of having to poll for new information you can use the Fortnox WebSocket API. 
+You start by creating a FortnoxWebSocketClient with your AccessToken and ClientSecret. You can then add the desired topics to your connection and once you're ready call `Connect` to initiate the connection.
+
+```CSharp
+var client = await new FortnoxWebSocketClient(this.connectionSettings.AccessToken, this.connectionSettings.ClientSecret)
+    .AddTopic(WebSocketTopic.Articles);
+```
+
+Once you're connected you can call `Listen` and specify your callback function, in order for this to not block your current thread you can run this within a Task like in the following snippet. Inside your listener callback you can make use of the `GetNextEvent` helper method to process messages as they come.
+
+```CSharp
+var task = new Task(async () =>
+{
+    try
+    {
+        (await client.Connect()).Listen(async (socket) =>
+        {
+            foreach (var response in client.GetNextEvent(socket))
```

Comment:
I think this requires additioal explanation, as it is not obvious what is happening here.

```CSharp
// GetNextEvent returns an enumeration and is iterated asyncrounusly as new events are yielded
foreach (var response in client.GetNextEvent(socket))
{
    if (response != null)
    {
        // Process messages here.
        return;
    }
}
```